### ISSUE TITLE
An error occurred! Can not decrypt this file, probably this is a shared file. [...] Problem with manually repaired OwnCloud after upgrade to 8.1.0

### ISSUE THREAD

**Issue by linoskoczek** [owncloud/core#18000]
### Steps to reproduce
No idea. I was just changing file structure to suit the info on the bottom of this guide (because migration script was causing problems, and it did not actually migrate, even though it finished): https://doc.owncloud.org/server/8.1/admin_manual/configuration_files/encryption_configuration.html#where-keys-are-stored
Everything seems to be in a good place (apart from *File keys for system-wide mount points* and *Share keys for files on a system-wide mount point*, because I didn't have them.) 

Whenever I try to open any file such error comes: 
*An error occurred! Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.*

#### Some of fle structure (maybe I did something wrong?) ####
/var/www/owncloud/data/user/files_encryption# ls
OC_DEFAULT_MODULE  keys

/var/www/owncloud/data/user/files_encryption/OC_DEFAULT_MODULE# ls
user.privateKey  user.publicKey

/var/www/owncloud/data/user/files_encryption/keys/dir_name/file_name/OC_DEFAULT_MODULE# ls
OC_DEFAULT_MODULE  fileKey  user.shareKey  recovery_abc123.shareKey

Note that there is OC_DEFAULT_MODULE again, because I had such weird file structure after using migration script:
*data/user/files_encryption/keys/files/files/file_path/OC_DEFAULT_MODULE/OC_DEFAULT_MODULE/fileKey*
I copied files from the last dir to the one before and removed */files/files/* (to make it look *normal*) for ownCloud. And I did it only to few files, just to test it. I still don't know how to remove this one, useless dir at the end of the path (in which the files are), but it doesn't matter ATM.

This is my Admin account. 

### Server configuration
**Operating system**:

Distributor ID: Ubuntu
Description:    Ubuntu 12.04.5 LTS
Release:        12.04
Codename:       precise
**Web server:**
Apache/2.2.22 (Ubuntu)
**Database:**
mysql  Ver 14.14 Distrib 5.5.44, for debian-linux-gnu (x86_64) using readline 6.2
**PHP version:**
PHP 5.4.43-1+deb.sury.org~precise+1 (cli)
**ownCloud version:** (see ownCloud admin page)
8.1.0.8
**Updated from an older ownCloud or fresh install:**
Upgraded from 8.0.5
**List of activated apps:**
 - activity: 2.0.1
 - calendar: 0.7.2
 - contacts: 0.4.0.0
 - encryption: 1.0.0
 - files: 1.1.9
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - gallery: 0.6.0
 - mail: 0.1.3
 - templateeditor: 0.1

**The content of config/config.php:**
<?php
$CONFIG = array (
  'instanceid' => 'xxx',
  'passwordsalt' => 'xxx',
  'trusted_domains' =>
  array (
    0 => 'xxx',
    1 => 'xxx',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'xxx',
  'dbtype' => 'mysql',
  'version' => '8.1.0.8',
  'dbname' => 'owncloud',
  'dbhost' => 'xxx',
  'dbtableprefix' => 'xxx',
  'dbuser' => 'xxx',
  'dbpassword' => 'xxx',
  'installed' => true,
  'forcessl' => true,
  'loglevel' => '4',
  'theme' => '',
  'maintenance' => false,
  'secret' => 'xxx',
  'forceSSLforSubdomains' => true,
  'mail_from_address' => 'noreply',
  'mail_smtpmode' => 'php',
  'mail_domain' => 'xxx',
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
No
**Are you using encryption:** yes/no
YES
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

#### Web server error log
No errors at this time.

#### ownCloud log (data/owncloud.log)
```
{"reqId":"xxx","remoteAddr":"xxx","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2015-07-30T19:12:22+00:00"} {"reqId":"xxx","remoteAddr":"xxx","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2015-07-30T19:12:22+00:00"} {"reqId":"xxx","remoteAddr":"xxx","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2015-07-30T19:12:23+00:00"} {"reqId":"xxx","remoteAddr":"xxx","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2015-07-30T19:12:24+00:00"} 
```

I will give every other info you need, just tell me.

**Comment by karlitschek** [comment]
@schiesbn

**Comment by mschaaf** [comment]
I have the same issue and can't access my files anymore. I need the system, so I have to revert the upgrade to the previous state.

**Comment by mschaaf** [comment]
The keymanager returns a empty key for the users. The call I talk about is on line 141 in the encryption php file.

$this->keyManager->getFileKey($this->path, $this->user);

path and user are correct set

**Comment by mschaaf** [comment]
In the method getFileKey the following variables are not initialized $encryptedFileKey and $shareKey so that thew method returns an empty string. Only the private key is set $privateKey

```
if ($encryptedFileKey && $shareKey && $privateKey) {
  return $this->crypt->multiKeyDecrypt($encryptedFileKey, $shareKey, $privateKey);
}
return '';
```

**Comment by mschaaf** [comment]
This file is expected ```/admin/files_encryption/keys/files/Notes/cvbcfb#fdgd.txt/OC_DEFAULT_MODULE/fileKey``` but is not. The part  ```OC_DEFAULT_MODULE/fileKey``` is missing. If I create this structure by hand is it expected to work then?

**Comment by mschaaf** [comment]
Generating OC_DEFAULT_MODULE/fileKey did not solve it. What I see from the path where it expects the key file is that it it misses the owncloud data folder as a prefix. I need to look inside the view->file_exists method do see if it is added there or not.

**Comment by mschaaf** [comment]
The file_exists seems to work but the path has another error it contains after the keys part a files folder and this did not exist. It seems that the files location for the encrypted files is taken as the real path. Can please someone from the owncloud team verify this?

To summarize:
Owncloud searches for a kefile and dind't find it. The path it looks at is ```/admin/files_encryption/keys/files/Notes/cvbcfb#fdgd.txt/OC_DEFAULT_MODULE/fileKey```
There are two problems in this path:
1. The last part did not exist /OC_DEFAULT_MODULE/fileKey
2. The ```files``` path did not exist.

The key file lays currently under ```/admin/files_encryption/keys/Notes/cvbcfb#fdgd.txt/fileKey```

**Comment by cmonteroluque** [comment]
@jvillafanez please see if you can replicate

**Comment by jvillafanez** [comment]
We have tests passing for ubuntu 14.04 + PHP 5.5.9 (the one that comes with that version), upgrading (tarball based) from 8.0.5 & 8.0.6 to the latest stable8.1; and the package upgrades (8.0.6 to 8.1.3, which is the latest 8.1 version available) works fine. I'm always talking about the basic installation.

> Updating \<files_versions\> ...
Updated \<files_versions\> to 1.0.6
Update 3rd-party app: calendar
Symfony\Component\Routing\Exception\RouteNotFoundException: Unable to generate a URL for the named route "calendar_index" as such route does not exist.
Update failed
Maintenance mode is kept active
ownCloud or one of the apps require upgrade - only a limited number of commands are available
Maintenance mode disabled

I'm worried about that error updating the calendar. My guess is that the update fails and the system is left in an inconsistent state, although I don't know how to reach that state.

**Comment by MorrisJobke** [comment]
> I'm worried about that error updating the calendar. My guess is that the update fails and the system is left in an inconsistent state, although I don't know how to reach that state.

see #19145

**Comment by MorrisJobke** [comment]
> I'm worried about that error updating the calendar. My guess is that the update fails and the system is left in an inconsistent state, although I don't know how to reach that state.

Should be solved by #19677 in 8.2.0

**Comment by DeepDiver1975** [comment]
@linoskoczek are you still seeing this issue? We are having issues to analyse the behavior - THX

**Comment by itsmortoncornelius** [comment]
I have the same issue on Debian GNU/Linux 7.8 (wheezy) after running the command "sudo -u www-data php occ upgrade" in my owncloud folder.
Running the script a second time causes the following message "ownCloud or one of the apps require upgrade - only a limited number of commands are available"

**Comment by cmonteroluque** [comment]
https://github.com/owncloud/core/issues/18000#issuecomment-148337752 is this resolved with 8.2.0?
